scripts/software/apps.sh

```diff
@@ -1,40 +1,243 @@
 #!/bin/bash
 
+# Check for an app and if it exists provide the version.
+function check() {
+    name=$1
+    version=$2
 
-# check for a app and if exists provide the version
-function check {
-	name=$1
-	version=$2
+    key=$name
 
-	# add the prefix app_ to every entry
-	key="app_$name"
-
-	if command -v "$name" > /dev/null; then
-		echo VALUE BAS "$key" \"$($name $version 2>&1 | sed '/./q; d')\" GROUP "apps"
-	else
-		echo VALUE BAS "$key" \""<not found>"\"	GROUP "missing_apps"
-	fi
+    if command -v "$name" >/dev/null; then
+        echo VALUE BAS "$key" \"$($name $version 2>&1 | sed '/./q; d')\" GROUP "apps"
+    fi
 }
 
+'''
+Shells.
+'''
 
 check "bash" "-version"
+check "zsh" "--version"
+check "csh" "--version"
+check "ksh" "--version"
+check "fish" "--version"
+
+'''
+Code Helpers.
+'''
+
+# Shell Parser, Formatter and Interpreter.
+check "shfmt" "--version"
+# Regex Evaluator.
+check "kiki" "--version"
+# Flash Card Tool to Help Memorize Shortcuts.
+check "mnemosyne" "--version"
+# Fix Previous Command.
+check "thefuck" "--version"
+# Hex Editor.
+check "ghex" "--version"
+# GUI Diff.
+check "meld" "--version"
+# Code Indexing.
+check "ctags" "--version"
+# Code Search (a specialized grep).
+check "ack" "--version"
+
+'''
+Terminal Management.
+'''
+
+# Terminal Multiplexer.
+check "byobu" "--version"
+# Another Terminal Multiplexer.
+check "tmux" "--version"
+# Multiple Terminals Management.
+check "terminator" "--version"
+
+'''
+# TODO:
+'''
+
+# Executes Commands in Makefile.
 check "make" "--version"
+
+'''
+Version Control.
+'''
+
 check "git" "--version"
+check "subversion" "--version"
+
+'''
+GNOME.
+'''
 
 check "gnome-software" "--version"
 
+'''
+KDE.
+'''
+
+check "kde-applications" "--version"
+# Bulk rename.
+check "krename" "--version"
+# File-version control.
+check "kdiff" "--version"
+
+'''
+Programming Languages & Compilers.
+'''
+
+# GNU Compiler Collection (C, C++, Go...).
 check "gcc" "--version"
 check "java" "-version"
 check "python" "-V"
+# Haskell Compiler.
+check "ghci" "--version"
 check "ruby" "-v"
+# Ruby version control.
+check "rvm" "--version"
+# Another Ruby Version Manager.
+check "rbenv" "--version"
 check "perl" "-v"
+check "lua" "-v"
 check "mysql" "--version"
+
+'''
+PenTest.
+'''
+
 check "freedom" "-version"
+# Security Scanner.
 check "nmap" "-V"
+# Wire protocols inspector.
+check "wireshark-cli" "--version"
+
+'''
+Remote Connectivity.
+'''
+
+check "ssh" "-V"
+# Uses FUSE to mount folders remotely using SSH.
+check "sshfs" "--version"
+# FTP Client.
+check "filezilla" "--version"
```

Comment:
>Reading locale option from /home/sima/.config/filezilla/filezilla.xml
FileZilla 3.40.0, compiled on 2019-01-25

We may need to use grep to get the correct line.

Reply:
https://stackoverflow.com/a/23702386/7303580

Reply:
Some pipes should do the trick:
**head -2 file | tail -1**                                       [FileZilla 3.40.0, compiled on 2019-01-25]
can also use awk, sed, cut and so on, for example:
**head -2 file | tail -1 | awk -F, '{ print $1 }'**      [FileZilla 3.40.0]

I'm leaving this up to you to decide on whichever suits best.

Reply:
I was thinking of using regex which matches only the version string in 'check'. I mean to apply this rule globally.

Reply:
This sounds like the best solution there is. It's the simplest and will ensure homomorphism.

Reply:
https://www.gnu.org/software/grep/manual/grep.html#index-_002dP

> PCRE support is here to stay

It'll be fine...

Reply:
`pcregrep -o "(\d+\.)+\d+"` if we want to be sure about compatibility.

Reply:
should I resolve the convos?

Reply:
Is it resolved?

Reply:
check commits

Reply:
I still get the first line on the output.

Reply:
Click 'Resolve conversation' and I will merge this.